File: 101-200/161-170/167-twoSum2/twoSum2.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int start = 1, end = numbers.size();

        while (start < end) {
            if (numbers[start-1] + numbers[end-1] > target)
                --end;
            else if (numbers[start-1] + numbers[end-1] < target)
                ++start;
            else
                return {start, end};
        }

        return vector<int> ();
    }
};

File: 101-200/161-170/167-twoSum2/twoSum2.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """

        dictionary = {}
        for i in range(len(numbers)):
            if numbers[i] in dictionary:
                return [dictionary[numbers[i]], i+1]
            else:
                dictionary[target-numbers[i]] = i+1

        return []